packages/xrpl/test/integration/transactions/offerCreate.test.ts

```diff
@@ -1,22 +1,31 @@
 import { assert } from 'chai'
 
-import { OfferCreate } from '../../../src'
+import { OfferCreate, TrustSet, Wallet } from '../../../src'
 import serverUrl from '../serverUrl'
 import {
   setupClient,
   teardownClient,
   type XrplIntegrationTestContext,
 } from '../setup'
-import { testTransaction } from '../utils'
+import {
+  testTransaction,
+  generateFundedWallet,
+  submitTransaction,
+} from '../utils'
 
 // how long before each test case times out
 const TIMEOUT = 20000
 
 describe('OfferCreate', function () {
   let testContext: XrplIntegrationTestContext
+  let wallet2: Wallet | undefined
 
   beforeEach(async () => {
     testContext = await setupClient(serverUrl)
+    if (!wallet2) {
+      // eslint-disable-next-line require-atomic-updates -- race condition doesn't really matter
+      wallet2 = await generateFundedWallet(testContext.client)
+    }
```

Comment:
Better practice would be to put this in a `beforeAll`

Reply:
If we use `beforeAll` instead of `beforeEach`, developers need to understand all the tests to be able to reason the success/failures of the tests. For instance, the number of objects in an account, the current balance in a wallet, etc will need to be understood for every test. Writing isolated tests will be more difficult. 

However, `beforeAll` could marginally improve the latency of the tests due to one-time initialization. What do you think ?

Reply:
Agreed - `beforeAll` and `beforeEach` aren't mutually exclusive.

Reply:
e942a4f9